plugins/inputs/nftables/nftables.go

```diff
@@ -0,0 +1,102 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux
+
+package nftables
+
+import (
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+var (
+	// defaultTables is the default nftables that we process
+	defaultTables = []string{"filter"}
+)
+
+type Nftables struct {
+	UseSudo bool     `toml:"use_sudo"`
+	Binary  string   `toml:"binary"`
+	Tables  []string `toml:"tables"`
+}
+
+func (*Nftables) SampleConfig() string {
+	return sampleConfig
+}
+
+func (nft *Nftables) Init() error {
+	if len(nft.Tables) == 0 {
+		return errors.New("invalid configuration - expected a 'Tables' entry with list of nftables to monitor")
+	}
+	return nil
+}
+
+func (nft *Nftables) Gather(acc telegraf.Accumulator) error {
+	for _, table := range nft.Tables {
+		err := nft.getTableData(table, acc)
+		if err != nil {
+			acc.AddError(err) // Continue through all tables
+		}
```

Comment:
The `acc.AddError` function already tests for the error so you can do

```suggestion
		acc.AddError(nft.getTableData(table, acc))
```

or even better add the error in the `getTableData` function.